### PR TITLE
Upgrade Cosmos SDK to v0.37.15

### LINKED ISSUE #53: Upgrade Cosmos SDK
Upgrade `medibloc/cosmos-sdk` from v0.34.7 to [v0.37.15](https://github.com/medibloc/cosmos-sdk/releases/tag/v0.37.15).

### PR DESCRIPTION
**Pull request by youngjoon-lee**
Closes #53 

## Changes

There are so many breaking changes when upgrading `medibloc/cosmos-sdk` from v0.34.7 to v0.37.15.

1. Most of boilerplate codes were moved into the cosmos-sdk. So, many Go packages were removed from the panacea-core.
    ```
    app/genesis.go
    app/invariants.go
    init/{collect.go, genesis_accts.go, gentx.go, ...}
    server/
    types/assets/
    version/
    ```
    - It also caused many changes in the `app/app.go` and `cmd/**/main.go`, but most of them are straight-forward fortunately.
    - Especially, the custom parameter definitions in the `app/genesis.go` were moved to the `medibloc/cosmos-sdk` repo. Please see https://github.com/medibloc/cosmos-sdk/pull/3.

2. The `x/**/module.go` was added.
    - For the latest cosmos-sdk, each module (`x/`) must implement `AppModuleBasic` and `AppModule`, so that they can be managed by `ModuleManager`. ([link](https://docs.staging-cosmos.network/master/building-modules/module-manager.html))

3. The `x/**/keeper.go` and some keeper-related codes were moved into the `x/**/keeper/`, according to [the new recommended directory structure](https://docs.staging-cosmos.network/master/building-modules/structure.html#structure).

4. The genesis-related types were moved into the `x/**/types/`,  according to [the new recommended directory structure](https://docs.staging-cosmos.network/master/building-modules/structure.html#structure).

## Tests

- Unit tests
- Running `panacead` from scratch
- AOL/DID operations

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -248,229 +282,44 @@ func NewPanaceaApp(
...
-	h.sh.BeforeDelegationRemoved(ctx, delAddr, valAddr)
-}
-
-// AfterDelegationModified nolint
-func (h StakingHooks) AfterDelegationModified(ctx sdk.Context, delAddr sdk.AccAddress, valAddr sdk.ValAddress) {
-	h.dh.AfterDelegationModified(ctx, delAddr, valAddr)
-	h.sh.AfterDelegationModified(ctx, delAddr, valAddr)
-}
+// ModuleAccountAddrs returns all the app's module account addresses.
+func (app *PanaceaApp) ModuleAccountAddrs() map[string]bool {
```
Comment (youngjoon-lee) [inline app/app.go:318]: 뭐하는 함수인지 정확히 다시 확인

File: app/app.go [hunk 2]
```diff
@@ -248,229 +282,44 @@ func NewPanaceaApp(
...
-		}
-	}
-
-	// assert runtime invariants
-	app.assertRuntimeInvariants()
-
-	return abci.ResponseInitChain{
-		Validators: validators,
-	}
+	return app.mm.InitGenesis(ctx, genesisState)
```
Comment (youngjoon-lee) [inline app/app.go:309]: 기존 코드와 동일한 일을 하는지 다시 확인

File: cmd/panacead/testnet.go [hunk 3]
```diff
@@ -186,81 +188,83 @@ func initTestnet(config *tmconfig.Config, cdc *codec.Codec) error {
...
 
-		coin := sdk.TokensFromTendermintPower(500) // TODO @ggomma check power - coin relation
-		accs = append(accs, app.GenesisAccount{
+		accTokens := sdk.TokensFromConsensusPower(1000)
+		accStakingTokens := sdk.TokensFromConsensusPower(500)
+		accs = append(accs, genaccounts.GenesisAccount{
 			Address: addr,
 			Coins: sdk.Coins{
-				sdk.NewCoin(assets.MicroMedDenom, coin),
+				sdk.NewCoin(fmt.Sprintf("%stoken", nodeDirName), accTokens),
```
Comment (youngjoon-lee) [inline cmd/panacead/testnet.go:200]: remove this line? what's the difference between accTokens and accStakingTokens?

File: cmd/panaceacli/main.go [hunk 4]
```diff
@@ -162,44 +120,35 @@ func txCmd(cdc *amino.Codec, mc []sdk.ModuleClients) *cobra.Command {
...
-func registerSwaggerUI(rs *lcd.RestServer) {
-	statikFS, err := fs.New()
-	if err != nil {
-		panic(err)
-	}
-	staticServer := http.FileServer(statikFS)
-	rs.Mux.PathPrefix("/swagger-ui/").Handler(http.StripPrefix("/swagger-ui/", staticServer))
+	client.RegisterRoutes(rs.CliCtx, rs.Mux)
+	authrest.RegisterTxRoutes(rs.CliCtx, rs.Mux)
+	app.ModuleBasics.RegisterRESTRoutes(rs.CliCtx, rs.Mux)
```
Comment (youngjoon-lee) [inline cmd/panaceacli/main.go:151]: why `client` and `authrest`?

Comment (youngjoon-lee) [reply cmd/panaceacli/main.go:151]: Due to the difference between `RegisterRoutes` and `RegisterTxRoutes`.

File: app/app.go (hunk 2 above)
Comment (youngjoon-lee) [reply app/app.go:309]: Done.

File: cmd/panacead/testnet.go (hunk 3 above)
Comment (youngjoon-lee) [reply cmd/panacead/testnet.go]: We don't need that. With that, the `genesis.json` for testnet is created like below, but I don't think we need the `node*token`.
```
    "accounts": [
      {
        "address": "panacea1gwrva7xvr38gugcfgtx4yz5uwddvwtymvp9rcg",
        "coins": [
          {
            "denom": "node0token",
            "amount": "1000000000"
          },
          {
            "denom": "umed",
            "amount": "500000000"
          }
        ],
```

File: app/app.go (hunk 1 above)
Comment (youngjoon-lee) [reply app/app.go:318]: See https://github.com/cosmos/cosmos-sdk/pull/4797. That PR was cherry-picked to the Gaia: https://github.com/cosmos/gaia/commit/1c8dad03ea4acdf66c3500cc9714ec293ab847dd

**Review by cl9200 — APPROVED** [review]
👍

**Comment by youngjoon-lee** [comment]
I also tested all commands in our manual. All works well. I'd say this PR is safe to be merged.

**Pull request merged**